packages/cli-plugin-metro/src/commands/start/runServer.ts

```diff
@@ -35,11 +46,70 @@ export type Args = {
   transformer?: string;
   watchFolders?: string[];
   config?: string;
+  terminal?: string;
   projectRoot?: string;
   interactive: boolean;
 };
 
+function logAlreadyRunningBundler(port: number) {
+  logger.info(`Metro Bundler is already for this project on port ${port}.`);
+}
+
+function logChangePortInstructions(port: number) {
+  logger.info(
+    `Please close the other packager running on port ${port}, or select another port with "--port".`,
+  );
+}
+
 async function runServer(_argv: Array<string>, ctx: Config, args: Args) {
+  let port = args.port ?? 8081;
+
+  if (args.terminal) {
+    startServerInNewWindow(port, ctx.root, ctx.reactNativePath, args.terminal);
+    return;
+  }
+
+  const packagerStatus = await isPackagerRunning(port);
+
+  const handleSomethingRunningOnPort = async () => {
+    const {change: kill} = await askForProcessKill(port);
+    if (kill) {
+      const pid = await getProcessIdFromPort(port);
+
+      if (pid) {
+        execa.sync('kill', [pid]);
+      }
```

Comment:
i think both `getProcessIdFromPort` and the following if statement should also cover running server on Windows

Reply:
correct! ✅